src/contexts/ThemeContext/ThemeContextProvide.tsx

```diff
@@ -0,0 +1,31 @@
+'use client';
+import { PropsWithChildren, useEffect, useState } from 'react';
+import { ThemeContext } from './ThemeContext';
+
+export function ThemeProvider({ children }: PropsWithChildren) {
+  const [theme, setTheme] = useState<'light' | 'dark' | 'auto'>('auto');
```

Comment:
Please use a `AppTheme[theme]` type here

Reply:
Also please use `matchMedia` to get the browser's theme value and use it as the default value